docs/book/user-guide/tutorials/trigger-pipelines-from-external-systems.md

```diff
@@ -0,0 +1,590 @@
+    api_key: str = Depends(get_api_key)
+):
+    """Trigger a ZenML pipeline with the given parameters."""
+    try:
+        # Dynamically import the pipeline
+        # your custom code here to load the source code of the pipeline
+        pipeline_func = import_pipeline(request.pipeline_name)
+        
+        # Configure the pipeline
+        if request.config_path:
+            configured_pipeline = pipeline_func.with_options(config_path=request.config_path)
+        else:
+            configured_pipeline = pipeline_func
+        
+        # Extract parameters from steps
+        step_parameters = {}
+        for step_name, step_config in request.steps.items():
+            if step_config.parameters:
+                step_parameters.update(step_config.parameters)
+        
+        # Run the pipeline with the provided parameters
+        run = configured_pipeline(**step_parameters)
+        
+        return {
+            "status": "success", 
+            "message": f"Pipeline '{request.pipeline_name}' triggered successfully",
+            "run_id": run.id if hasattr(run, 'id') else None
+        }
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Failed to trigger pipeline: {str(e)}")
+
+@app.post("/trigger-template")
+async def trigger_template(
+    request: TemplateRunRequest,
+    api_key: str = Depends(get_api_key)
+):
+    """Trigger a pipeline using a run template."""
+    try:
+        # Get the template
+        template = Client().get_run_template(request.template_id)
+        
+        # Get the template's configuration
+        config = template.config_template or {}
+        
+        # Update the configuration with step parameters
+        config["steps"] = {
+            step_name: {"parameters": step_config.parameters}
+            for step_name, step_config in request.steps.items()
+        }
+        
+        # Trigger the pipeline with the updated configuration
+        run = Client().trigger_pipeline(
+            template_id=template.id,
+            run_configuration=PipelineRunConfiguration(**config),
+        )
+        
+        return {
+            "status": "success",
+            "message": f"Pipeline triggered successfully from template '{template.name}'",
+            "run_id": run.id,
+            "template_id": template.id,
+            "template_name": template.name
+        }
+    except Exception as e:
+        raise HTTPException(
+            status_code=500,
+            detail=f"Failed to trigger template: {str(e)}"
+        )
+```
+
+### Deploying Your Custom API
+
+Deploy the API service to make it accessible to external systems:
+
+```bash
+# Install required dependencies
+pip install fastapi uvicorn
+
+# Set a secure API key
+export PIPELINE_API_KEY="your-secure-api-key-here"
+
+# Start the API server
+python pipeline_api.py
+```
+
+For production, consider deploying as a containerized service:
+
+```dockerfile
+FROM python:3.9-slim
```

Comment:
doesnt this need zenml installed?

Reply:
Yes. It specifies a requirements.txt file where you'd have that all set up. I'll
add a note.